contracts/strategies/LPOptimiserStrategy.sol

```diff
@@ -124,49 +149,54 @@ contract LPOptimiserStrategy is DefaultAccessControl, ILpCallback {
     /// @notice Convert a fixed rate to a tick in wad
     /// @param fixedRateWad The fixed rate to be converted to a tick in wad
     /// @return int256 The tick in wad
-    function convertFixedRateToTick(int256 fixedRateWad) public view returns (int256) {
+    function convertFixedRateToTick(int256 fixedRateWad) public pure returns (int256) {
         return -PRBMathSD59x18.div(PRBMathSD59x18.log2(int256(fixedRateWad)), PRBMathSD59x18.log2(int256(LOG_BASE)));
     }
 
     function convertTickToFixedRate(int24 tick) public pure returns (uint256) {
+        // 1. Convert the tick into X96 sqrt price (scaled by 2^96)
         uint160 sqrtPriceX96 = TickMath.getSqrtRatioAtTick(tick);
 
+        // 2. Convert the X96 sqrt price (scaled by 2^96) to wad 1/sqrt price (scaled by 10^18)
         uint256 sqrtRatioWad = FullMath.mulDiv(1e18, FixedPoint96.Q96, sqrtPriceX96);
 
+        // 3. Convert 1/sqrt price into fixed rate (1/price) 
         uint256 fixedRateWad = sqrtRatioWad.mul(sqrtRatioWad);
+
+        // 4. Return the fixed rate
         return fixedRateWad;
     }
 
     /// @notice Set new optimal tick range based on current twap tick given that we are using the offchain moving average of the fixed rate in the current iteration
     /// @param currentFixedRateWad currentFixedRate which is passed in from a 7-day rolling avg. historical fixed rate.
     /// @return newTickLower The new lower tick for the rebalanced position
     /// @return newTickUpper The new upper tick for the rebalanced position
-    function rebalanceTicks(uint256 currentFixedRateWad) public returns (int24 newTickLower, int24 newTickUpper) {
+    function rebalanceTicks(uint256 index, uint256 currentFixedRateWad) public returns (int24 newTickLower, int24 newTickUpper) {
         _requireAtLeastOperator();
-        require(rebalanceCheck(currentFixedRateWad), ExceptionsLibrary.REBALANCE_NOT_NEEDED);
+        require(rebalanceCheck(index, currentFixedRateWad), ExceptionsLibrary.REBALANCE_NOT_NEEDED);
 
-        uint256 sigmaWad = _sigmaWad;
+        VaultParams memory vaultParams = _vaultParams[index];
+        IVoltzVault vault = _vaults[index];
 
         // 0. Get tickspacing from vamm
-        int24 tickSpacing = _vamm.tickSpacing();
+        int24 tickSpacing = vault.vamm().tickSpacing();
```

Comment:
shall we store another array with corresponding VAMMs?

Reply:
So that we don't need to call vault.vamm() every time we rebalance ticks (although it will not happen to often, and gas is supported by us)